presto-main/src/main/java/com/facebook/presto/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -386,6 +395,11 @@ private Type handleResolvedField(Expression node, FieldId fieldId, Type resolved
                 }
             }
 
+            Field field = baseScope.getRelationType().getFieldByIndex(fieldId.getFieldIndex());
```

Comment:
I think it would be better to modify `handleResolvedField(.., ResolvedField, ..)`, since then you have direct access to `resolvedField.getField()`

Reply:
That was my first thought, but FieldReference doesn't go through that path, so I would have to duplicate the logic for visitFieldReference.